pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -717,6 +717,11 @@ func (admitter *VMsAdmitter) shouldAllowCPUHotPlug(vm *v1.VirtualMachine) error
 		!vmi.Status.MigrationState.Completed {
 		return fmt.Errorf("cannot update CPU sockets while VMI migration is in progress")
 	}
+
+	err = EnsureNoMigrationConflict(admitter.VirtClient, vm.Name, vm.Namespace)
```

Comment:
Isn't this still racy? While it may be true that this returns nil another component or even the user might post a migration at the same time. I am not sure I understand this change. As suggested in the past I would just make sure the code around hotplug is resilient to this situation rather than prevent it from happening.

Reply:
This change makes sure that changing the topology will NOT be allowed when a migration object for a given VMI exists.

If the migration is posted *After*, the hot plug has changed the topology - there is no problem. The target pod will be created with the altered topology and correct resources for hot plug.
The only issue right now that exists (without this PR) is when a competing migration is posted before the target pod handoff and this PR blocks it.

I'm 100% sure what resiliency are you suggesting, however, we can continue this work in a follow-up. This PR only prevents initiating the CPU hot plug when there is another inflight migration. - Same way as double migration is not allowed - hence using the same function.

Reply:
All I am saying is that both the migration and the change of the spec seem to be able l go through. Certainly, the gap is being closed but not entirely.

My suggestion would be something along these lines(Warning, I did not put time into this), a vmim is being annotated with cpus that were used to calculate the target Pod. After the migration, the cpus are propagated to VMI and handler can use this in order to determine if hotplug is safe to do/or at least partial hotplug as you then don't need to reject changes.

Reply:
@xpivarc This needs additional thinking, from my POV. I don't see how this belongs to this PR. 
For a non-dedicated CPUs case, a lot can change in between the annotation of the VMIM and the actual plug. For the dedicated CPUs case, what we have right now is close to being enough.
I don't see a reason to hold up this PR. 
We should definitely continue the discussion, but this PR solves specific immediate issues.